### PR TITLE
v1.10: Bump lru crate

### PR DESCRIPTION
**Pull request by CriesofCarrots**
https://github.com/solana-labs/solana/pull/24150 for v1.10

### CODE REVIEW

File: gossip/src/cluster_info_metrics.rs [hunk 1]
```diff
@@ -536,13 +536,11 @@ pub(crate) fn submit_gossip_stats(
         .pull
         .votes
         .into_iter()
-        .map(|(slot, num_votes)| (*slot, *num_votes))
```
Comment (KirillLykov) [inline gossip/src/cluster_info_metrics.rs:539]: Is this change related to lru? looks a bit confusing

Comment (CriesofCarrots) [reply gossip/src/cluster_info_metrics.rs:539]: Yes, the type of `CrdsDataStats.votes` is `LruCache<Slot, usize>`
